### PR TITLE
cudf-udf integration test against python3.9 [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
To fix https://github.com/NVIDIA/spark-rapids/issues/7773

RAPIDS 23.06 drops python 3.8 packages and supports python3.9 and 3.10

### CODE REVIEW

**Comment by pxLi** [comment]
as it touchs testing scripts, we cannot skip ci here

**Review by pxLi — DISMISSED** [review]

File: jenkins/databricks/init_cudf_udf.sh [hunk 1]
```diff
@@ -20,16 +20,23 @@
...
 export PYSPARK_PYTHON=${PYSPARK_PYTHON:-"$(which python)"}
 PYTHON_VERSION=$(${PYSPARK_PYTHON} -c 'import sys; print("{}.{}".format(sys.version_info.major, sys.version_info.minor))')
-# cudf 23.02+ do not support python 3.9. ref: https://docs.rapids.ai/notices/rsn0022/
-[[ "$PYTHON_VERSION" == '3.9' ]] && PYTHON_VERSION='3.8'
+# Rapids 23.06+ drops python 3.8 conda packages. ref: https://docs.rapids.ai/notices/rsn0029/
+if [[ "$(printf '%s\n' "3.9" "${PYTHON_VERSION}" | sort -V | head -n1)" = "3.9" ]]; then
+    # To fix "'lsb_release -a' returned non-zero". ref: https://github.com/pypa/pip/issues/4924
+    [[ -n "$(which lsb_release)" ]] && mv $(which lsb_release) $(which lsb_release)"-bak"
+else
+    # Set python versiomn to make it the same for both shell mode and batch mode
```
[the remark this answers is not in the document]
Comment (NvTimLiu) [reply jenkins/databricks/init_cudf_udf.sh]: Oh this notation is not needed, we already set the python version to 3.9  for both batch and shell mode here, when running cudf-udf tests: 

https://github.com/NVIDIA/spark-rapids/blob/branch-23.06/jenkins/databricks/cudf_udf_test.sh#L45-L46

**Comment by NvTimLiu** [comment]
build

**Review by pxLi — APPROVED** [review]

**Comment by pxLi** [comment]
Going to merge this one. Lets keep monitoring nightly runs status

**Pull request merged**